relay/server.go

```diff
@@ -408,8 +439,144 @@ func getKeysFromChunkRequest(request *pb.GetChunksRequest) ([]v2.BlobKey, error)
 	return keys, nil
 }
 
-// gatherChunkDataToSend takes the chunk data and narrows it down to the data requested in the GetChunks request.
-func gatherChunkDataToSend(
+// Used to pass status of downloads from goroutines up to controlling function.
+type downloadResult struct {
+	key   v2.BlobKey
+	found bool
+	err   error
+}
+
+// Download and compile the chunk data to send back to the client.
+func (s *Server) gatherChunkDataToSend(
```

Comment:
this new method is very long and hard to read. Could it be broken up into more manageable parts?